packages/video_player/README.md

```diff
@@ -64,17 +64,12 @@ class _VideoAppState extends State<VideoApp> {
       'http://www.sample-videos.com/video/mp4/720/big_buck_bunny_720p_20mb.mp4',
     )
       ..addListener(() {
-        final bool isPlaying = _controller.value.isPlaying;
-        if (isPlaying != _isPlaying) {
-          setState(() {
-            _isPlaying = isPlaying;
-          });
-        }
-      })
-      ..initialize().then((_) {
+        print('completed listener: $_controller');
```

Comment:
It seems like this is being changed because you're getting rid of the initialized message - is this really an improvement though?

Reply:
When the future of `..initialize()` completes, the video player may not actually be initialized yet (it generally isn't). In that case, the following code is a NOP as the 1st frame of the video is not yet available:

```DART
..initialize().then((_) {
        // Ensure the first frame is shown after the video is initialized, even before the play button has been pressed.
        setState(() {});
      });
```

So I replaced the `..addListener(listener)` callback by the `then()` listener, which guarantees the 1st frame of the video is shown (see https://github.com/flutter/plugins/blob/master/packages/video_player/lib/video_player.dart#L233). This has the added benefit of simplifying the code as the `_isPlaying` variable is no longer needed.

Some plugin exceptions were being lost before, so I've  added a catchError callback to notify the user.